app/controllers/sequences_controller.rb

```diff
@@ -34,9 +34,14 @@ def show
     current_project
     respond_to do |format|
       format.html do
+        if @sequence_run && !params[:sequence_run_key]
+          redirect_to sequence_with_sequence_run_key_path(@sequence, @sequence_run.key, params)
+          return
+        end
         if @sequence_run && @sequence_run.has_been_run
           unless params[:show_index]
-            redirect_to sequence_activity_path(@sequence, @sequence_run.most_recent_activity)
+            activity = @sequence_run.most_recent_activity
+            redirect_to sequence_activity_with_run_path(@sequence, activity, @sequence_run.run_for_activity(activity))
             return
           end
         end
```

Comment:
Is there a valid case where `@sequence_run` can be nil here? If not perhaps we should show an error.